### PR TITLE
woof: init at 2012-05-31

### PR DESCRIPTION
**Pull request by lschuermann**
###### Motivation for this change
`woof` (Web Offer One File) is a small and simple command line utility written in Python that makes file sharing in a local network easy.
The most basic usecase is to just start it with a file as an argument, which it then serves via an HTTP web-server. Additionally, it is able to compress and serve a folder, provide an upload form, or even offer itself using command line switches.

This is a very useful utility I would not like to miss on any portable computers.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Note that when `woof` is used with the `-s` option (serving itself), because of how the package gets built, it offers the Python file with the path to the Python interpreter hardcoded inside the Nix-store. I'd accept that as a compromise, as I think that fixing this would either require changing the `woof` source itself or the way Nix packages get built. Also, people wanting to get `woof` should use the awesome Nix package-manager anyway. ;)

Is it possible for me to become a maintainer for this package? If so, what steps should I follow? I sadly can't remember all of the awesomeness that was shown to me by the NixOS assembly at 34C3.

### CODE REVIEW

File: pkgs/tools/misc/woof/default.nix [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      cp $src $out/bin/woof
+      chmod +x $out/bin/woof
+    '';
+
+  meta = {
+    homepage = http://www.home.unix-ag.org/simon/woof.html;
+    description = "Web Offer One File - Command-line utility to easily exchange files over a local network";
+    license = stdenv.lib.licenses.gpl2Plus;
+    platforms = stdenv.lib.platforms.unix;
+  };
```
Comment (dotlambda) [inline pkgs/tools/misc/woof/default.nix]: To add yourself as a maintainer, use
```
maintainers = with stdenv.lib.maintainers; [ lschuermann ];
```
And don't forget to add yourself to https://github.com/NixOS/nixpkgs/blob/master/lib/maintainers.nix

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/woof/default.nix (hunk 1 above)
Comment (dotlambda) [reply pkgs/tools/misc/woof/default.nix]: If you want to leave out all that `stdenv.lib`, you can do it like that:
```
meta = with stdenv.lib; {
  homepage = http://www.home.unix-ag.org/simon/woof.html;
  description = "Web Offer One File - Command-line utility to easily exchange files over a local network";
  license = licenses.gpl2Plus;
  platforms = platforms.unix;
  maintainers = with maintainers; [ lschuermann ];
};

File: pkgs/tools/misc/woof/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  version = "2012-05-31";
+  name = "woof-${version}";
+  
+  src = fetchurl {
+    url = "http://www.home.unix-ag.org/simon/woof-${version}.py";
+    sha256 = "d84353d07f768321a1921a67193510bf292cf0213295e8c7689176f32e945572";
+  };
+
+  buildInputs = [ python ];
+  runtimeDeps = [ python ];
```
Comment (dotlambda) [inline pkgs/tools/misc/woof/default.nix]: No need for `runtimeDeps`.

**Review by dotlambda — CHANGES_REQUESTED** [review]

File: pkgs/tools/misc/woof/default.nix (hunk 2 above)
Comment (dotlambda) [reply pkgs/tools/misc/woof/default.nix]: Actually, I've never seen `runtimeDeps` before.
The manual has a nice overview: https://nixos.org/nixpkgs/manual/#ssec-stdenv-attributes
Essentially, `buildInputs` are for run-time dependencies whereas `nativeBuildInputs` are for build-time dependencies.

**Review by dotlambda — APPROVED** [review]

**Comment by dotlambda** [comment]
Maybe @FRidh can take a look at this?

**Comment by adisbladis** [comment]
Pushed in https://github.com/NixOS/nixpkgs/commit/a6dca0427221d7c249a9b6f1581cf0d73baf51da

Thanks :)

**Pull request closed without merging**